### ISSUE TITLE
No Route To Host / Connection Issues

### ISSUE THREAD

**Issue by rossedman** [ansible/awx#53]
### Summary

It appears awx_task runner can't connect to anything. I am assuming this has something to do with docker bridge networking or some other networking component. The node this is deployed to has public internet access as well as private network access. 

### Environment

Ubuntu 16.04
AWX Deployment Script Used

```
9806b36e4525        awx_web:1.0.0.286    "/tini -- /bin/sh ..."   About an hour ago   Up 2 seconds        0.0.0.0:80->8052/tcp                 awx_web
d9fd5cce6a40        awx_task:1.0.0.286   "/tini -- /bin/sh ..."   2 hours ago         Up 2 hours          8052/tcp                             awx_task
87b101920dcf        memcached:alpine     "docker-entrypoint..."   2 hours ago         Up 2 hours          11211/tcp                            memcached
37504846c7f5        rabbitmq:3           "docker-entrypoint..."   2 hours ago         Up 2 hours          4369/tcp, 5671-5672/tcp, 25672/tcp   rabbitmq
bbf46397d84b        postgres:9.6         "docker-entrypoint..."   2 hours ago         Up 2 hours          5432/tcp                             postgres
```

### Steps To Reproduce:

When running a template job in containers every run returns. Used Install guide on fresh Ubuntu 16.04 node. Tried overwriting `ansible_host` and other parameters.

```
Identity added: /tmp/awx_21_6d2f8b/credential_7 (/tmp/awx_21_6d2f8b/credential_7)
Using /etc/ansible/ansible.cfg as config file
ERROR! Attempted to execute "/usr/lib/python2.7/site-packages/awx/plugins/inventory/awxrest.py" as inventory script: Inventory script (/usr/lib/python2.7/site-packages/awx/plugins/inventory/awxrest.py) had an execution error: HTTPConnectionPool(host='awxweb', port=8052): Max retries exceeded with url: /api/v1/inventories/5/script/?hostvars=1 (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x205e9d0>: Failed to establish a new connection: [Errno 113] No route to host',))
None
```

### Expected Results:

Be able to connect to hosts or receive enough output to debug.

### Actual Results:

Posted above.

**Comment by kareiva** [comment]
Are you behind multiple network layers? I had similar issues with deploying to OpenStack instances. Just a wild-ass guess, but try reducing the MTU in the docker daemon systemd configuration. 

https://rahulait.wordpress.com/2016/02/28/modifying-default-mtu-for-docker-containers/

**Comment by rossedman** [comment]
@kareiva I think this may have something to do with docker bridge network. I believe the task runner is the service doing the configuring. So while I can access AWX and save things to the database, there still needs to be a way to expose the task runner.

**Comment by leifmadsen** [comment]
I ran into this, and basically, you need to connect via the hard coded `awxweb` hostname (add to your `/etc/hosts` file) and then nginx tends to route things properly. This is what I ended up doing after opening port 80 in a security group in my OpenStack deployment.

Probably related to #99 partially. The whole `settings.py` needs to be templated so that we can adjust the ports and hostnames where things are directed. You'll notice that the awx_web host in Docker is pointing external port 80 to internal port 8052 (which is not available externally).

**Comment by rossedman** [comment]
@leifmadsen will give this a shot thanks!

**Comment by matburt** [comment]
This is probably fixed in  #162
